src/System.Binary.Base64/System/Binary/Base64.cs

```diff
@@ -135,25 +142,37 @@ public static int Encode(ReadOnlySpan<byte> source, Span<byte> destination)
             while (sourceIndex < srcLength - 2)
             {
                 result = Encode(ref Unsafe.Add(ref srcBytes, sourceIndex));
+                if (destIndex > destLength - 4) goto FalseExit;
```

Comment:
I don't think it will hurt performance. The check will happen regardless, so it is better to exit sooner if we don't have room. I will fix this.